test/cpp/jit/test_gpu.cpp

```diff
@@ -5598,6 +5599,8 @@ void testGPU_FusionSmemReduce() {
       aten_output.allclose(outputs[0], 1e-5, 1e-5),
       "Error of: ",
       aten_output.sub(outputs[0]).abs().max());
+  TORCH_CHECK(fe.summary().war_hazard_syncs.size() == 1);
```

Comment:
The idea looks good to me, and this is fine for this PR (modulo the `FusionExecutor::summary()` comment) but I'd like to revisit the approach in the near future so validation is strictly outside the Kernel object.

The `KernelSummary` should provide information required for compilation & execution. Extra validation should be optional, not on the common path.

Reply:
Maybe a `DebugKernelSummary` that we can generate from the cpp test via  `FusionExecutor::kernel()`